--- a/kernel-build.eclass
+++ b/kernel-build.eclass
@@ -206,8 +206,8 @@
 kernel-build_src_test() {
 	debug-print-function ${FUNCNAME} "${@}"
 	local targets=( modules_install )
-	# on arm or arm64 you also need dtb
-	if use arm || use arm64 || use riscv; then
+### SWITCH: l4t sources does not support that
+	if use riscv; then
 		targets+=( dtbs_install )
 	fi
 
@@ -234,8 +234,8 @@
 	# do not use 'make install' as it behaves differently based
 	# on what kind of installkernel is installed
 	local targets=( modules_install )
-	# on arm or arm64 you also need dtb
-	if use arm || use arm64 || use riscv; then
+### SWITCH: l4t sources does not support that
+	if use riscv; then
 		targets+=( dtbs_install )
 	fi
 
