--- /var/db/repos/gentoo/eclass/kernel-build.eclass	2023-12-11 14:10:35.000000000 +0100
+++ eclass/kernel-build.eclass	2023-12-15 12:25:02.958089409 +0100
@@ -257,8 +257,8 @@
 	# do not use 'make install' as it behaves differently based
 	# on what kind of installkernel is installed
 	local targets=( modules_install )
-	# on arm or arm64 you also need dtb
-	if use arm || use arm64 || use riscv; then
+	# on arm or arm64 you also need dtb - But switch l4t sources does not support that
+	if use riscv; then
 		targets+=( dtbs_install )
 	fi
 
