--- a/kernel-build.eclass
+++ b/kernel-build.eclass
@@ -142,10 +142,11 @@
 kernel-build_src_test() {
 	debug-print-function ${FUNCNAME} "${@}"
 	local targets=( modules_install )
-	# on arm or arm64 you also need dtb
-	if use arm || use arm64; then
-		targets+=( dtbs_install )
-	fi
+### SWITCH: l4t sources does not support that
+#	# on arm or arm64 you also need dtb
+#	if use arm || use arm64; then
+#		targets+=( dtbs_install )
+#	fi
 
 	emake O="${WORKDIR}"/build "${MAKEARGS[@]}" \
 		INSTALL_MOD_PATH="${T}" "${targets[@]}"
@@ -170,10 +171,11 @@
 	# do not use 'make install' as it behaves differently based
 	# on what kind of installkernel is installed
 	local targets=( modules_install )
-	# on arm or arm64 you also need dtb
-	if use arm || use arm64; then
-		targets+=( dtbs_install )
-	fi
+### SWITCH: l4t sources does not support that
+#	# on arm or arm64 you also need dtb
+#	if use arm || use arm64; then
+#		targets+=( dtbs_install )
+#	fi
 
 	emake O="${WORKDIR}"/build "${MAKEARGS[@]}" \
 		INSTALL_MOD_PATH="${ED}" INSTALL_PATH="${ED}/boot" "${targets[@]}"
