--- sys-kernel/installkernel/installkernel-29.ebuild
+++ sys-kernel/installkernel/installkernel-29.ebuild
@@ -17,12 +17,15 @@
 LICENSE="GPL-2+"
 SLOT="0"
 KEYWORDS="~arm64"
-IUSE="dracut efistub grub refind systemd systemd-boot uki ukify"
+IUSE="dracut efistub +nsw grub refind systemd systemd-boot uki ukify"
+
+PATCHES=( "${FILESDIR}"/add-nsw-install-18.patch )
+
 REQUIRED_USE="
 	efistub? ( systemd )
 	systemd-boot? ( systemd )
 	ukify? ( uki )
-	?? ( efistub grub refind systemd-boot )
+	?? ( efistub grub refind systemd-boot nsw )
 "
 # efistub requires systemd's kernel-install because:
 # - We need support for removal to clean-up the created entry
@@ -33,6 +36,8 @@
 	!<=sys-kernel/installkernel-systemd-3
+	nsw? ( >=sys-libs/gentoo-config-files-20240127-r1[installkernel] )
 	dracut? (
 		>=sys-kernel/dracut-060_pre20240104-r4
+		sys-libs/gentoo-config-files[initramfs]
 		uki? (
 			|| (
 				sys-apps/systemd[boot(-)]
