--- x11-libs/libdrm/libdrm-2.4.111.ebuild
+++ x11-libs/libdrm/libdrm-2.4.111.ebuild
@@ -38,4 +38,6 @@
 BDEPEND="${PYTHON_DEPS}
 	$(python_gen_any_dep 'dev-python/docutils[${PYTHON_USEDEP}]')"
 
+PATCHES=( "${FILESDIR}"/libdrm.patch )
+
 python_check_deps() {
@@ -63,3 +65,10 @@
 	)
 	meson_src_configure
 }
+
+multilib_src_install() {
+	meson_src_install
+	if use video_cards_tegra; then
+		rm ${D}/usr/lib64/libdrm.so.2
+	fi
+}

