--- a/linux-firmware-99999999.ebuild
+++ b/linux-firmware-99999999.ebuild
@@ -29,7 +29,7 @@
 		linux-fw-redistributable ( BSD-2 BSD BSD-4 ISC MIT no-source-code ) )
 	unknown-license? ( all-rights-reserved )"
 SLOT="0"
-IUSE="initramfs +redistributable savedconfig unknown-license"
+IUSE="initramfs +redistributable savedconfig unknown-license +switch_mod"
 REQUIRED_USE="initramfs? ( redistributable )"
 
 RESTRICT="binchecks strip test
@@ -256,6 +256,11 @@
 		IFS="${OLDIFS}"
 	fi
 
+	if use switch_mod; then
+		einfo "Removing files provided by Switchroot kernel ..."
+		sed -i /brcmfmac4356-pcie.bin/d  WHENCE || die
+	fi
+
 	restore_config ${PN}.conf
 }
 
