diff -Naur libdrm-2.4.110/meson.build libdrm-2.4.110-2/meson.build
--- libdrm-2.4.110/meson.build	2022-04-28 01:23:41.595660263 +0200
+++ libdrm-2.4.110-2/meson.build	2022-04-28 01:44:50.085436924 +0200
@@ -294,6 +294,7 @@
   output : 'config.h',
 )
 add_project_arguments('-include', '@0@'.format(config_file), language : 'c')
+add_global_link_arguments('-Wl,-Bsymbolic-functions', '@0@'.format(config_file), language : 'c')
 
 inc_root = include_directories('.')
 inc_drm = include_directories('include/drm')
