--- installkernel/installkernel
+++ installkernel/installkernel
@@ -363,6 +363,25 @@
 			echo "Installing Unified Kernel Image for ${ver}..."
 		fi
 		updatever "${img_dest}" "${uki}" "${suffix}"
+	# The NSW/switchroot way
+	elif [ -f "${INSTALLKERNEL_STAGING_AREA}/nx-plat.dtimg" ] && [ "${INSTALLKERNEL_LAYOUT}" = nsw ]; then
+		echo "Installing nx-plat.dtimg for ${ver}..."
+		updatever nx-plat "${INSTALLKERNEL_STAGING_AREA}/nx-plat.dtimg" .dtimg
+
+		echo "Installing uImage for ${ver}..."
+		updatever uImage "${INSTALLKERNEL_STAGING_AREA}/uImage"
+
+		if [ -f "${INSTALLKERNEL_STAGING_AREA}/initramfs" ]; then
+			echo "Installing initramfs image for ${ver}..."
+			updatever initramfs "${INSTALLKERNEL_STAGING_AREA}/initramfs"
+		fi
+
+		config=$(dirname "${map}")
+		config="${config}/.config"
+		if [ -f "${config}" ]; then
+			echo "Installing config for ${ver}..."
+			updatever config "${config}"
+		fi
 	else
 		if [ "${INSTALLKERNEL_VERBOSE}" -gt 0 ]; then
 			echo "Installing kernel image for ${ver}..."
