--- installkernel/installkernel
+++ installkernel/installkernel
@@ -100,6 +100,13 @@
 if [ -f "${uki}" ]; then
 	suffix=.efi
 	updatever ${img_dest} "${uki}" ${suffix}
+elif  [ -f "${base_dir}/nx-plat.dtimg" ]; then
+	updatever nx-plat "${base_dir}/nx-plat.dtimg" .dtimg
+	updatever uImage "${base_dir}/uImage"
+	initramfs="${base_dir}/initramfs"
+	if [ -f "${initramfs}" ]; then
+		updatever initramfs "${initramfs}"
+	fi
 else
 	suffix=
 	updatever ${img_dest} "${img}"
